doc/managing_pachyderm/data_management.md

```diff
@@ -0,0 +1,47 @@
+# Data Management Best Practices
+
+This document discusses best practices for minimizing the space needed to store your Pachyderm data, increasing the performance of your data processing as related to data organization, and general good ideas when you are using Pachyderm to version/process your data.
+
+- [Shuffling files](#shuffling-files)
+- [Garbage collection](#garbage-collection)
+- [Setting a root volume size](#setting-a-root-volume-size)
+
+## Shuffling files
+
+Certain pipelines simply shuffle files around (e.g., organizing files into buckets).  If you find yourself writing a pipeline that does a lot of copying, such as [Time Windowing](http://docs.pachyderm.io/en/latest/cookbook/time_windows.html), it probably falls into this category.
+
+The best way to shuffle files, especially large files, is to create **symlinks** in the output directory that point to files in the input directory.
+
+For instance, to move a file `log.txt` to `logs/log.txt`, you might be tempted to write a [`transform`](http://pachyderm.readthedocs.io/en/latest/reference/pipeline_spec.html#transform-required) like this:
+
+```sh
+cp /pfs/input/log.txt /pfs/out/logs/log.txt
+```
+
+However, it's more efficient to create a symlink:
+
+```sh
+ln -s /pfs/input/log.txt /pfs/out/logs/log.txt
+```
+
+Under the hood, Pachyderm is smart enough to recognize that the output file simply symlinks to a file that already exists in Pachyderm, and therefore skips the upload altogether.
+
+Note that if your shuffling pipeline only needs the names of the input files but not their content, you can use [`lazy input`](http://pachyderm.readthedocs.io/en/latest/reference/pipeline_spec.html#atom-input).  That way, your shuffling pipeline can skip both the download and the upload.
+
+## Garbage collection
+
+When a file/commit/repo is deleted, the data is not immediately removed from the underlying storage system (e.g. S3) for performance and architectural reasons.  This is similar to how when you delete a file on your computer, the file is not necessarily wiped from disk immediately.
+
+To actually remove the data, you may need to manually invoke garbage collection.  The easiest way to do it is through `pachctl garbage-collect`.  Currently `pachctl garbage-collect` can only be started when there are no active jobs running.  You also need to ensure that there's no ongoing `put-file`.  Garbage collection puts the cluster into a readonly mode where no new jobs can be created and no data can be added.
+
+## Setting a root volume size
+
+When planning and configuring your Pachyderm deploy, you need to make sure that each node's root volume is big enough to store the biggest "datum" you expect to process anywhere on your DAG plus the size of the output files that will be written for that datum.
```

Comment:
This is a start ... but could still lead you to set the `rootVolumeSize` too low ... e.g. if you have 10 datums getting processed at once (within this pipeline or across pipelines) you'd run out of space. This is kinda tricky. We don't have a great way for users to measure this right now (but this kicked off an interesting discussion about how we could present this on the UI) ... so it's a bit of trial and error. 

I think the todo here is just to make them aware that this is the total processing bandwidth for all their jobs that are running at any point in time. So do the calculation for each type of job, then do something like:

```
fileStorageNeededPerDatum x numberOfDatumsRunningSimultaneously / numberOfNodes
```
Note - the numberOfDatumsRunningSimultaneously could be within/across pipelines.

That's crude, but an ok place to start. It's reasonable to start w a value and up it later as your needs increase.